### ISSUE TITLE
Proposal: allow `endPaint` to accept a `customProperties` object to do custom reporting

### ISSUE THREAD

**Issue by dmatteo** [Zizzamia/perfume.js#68]
The current API for `endPaint` only allows a metric name to be passed in, which is limiting the potential to report additional properties, especially when using custom analytics tracking.

The proposed syntax will accept the object, and surface it in logging statements and analytics tracking, e.g.

**on the tracking side**
```js
perfume.endPaint(METRIC_NAME, { page: '/login' }
```

**on the reporting site**
```js
export const perfumeOptions = {
    firstContentfulPaint: true,
    firstInputDelay: true,
    analyticsTracker: (metricName: string, duration: number, browser?: *, customProperties: { [key:string]: string }) => {
        track(`custom_tracking_${metricName}`, {
            duration,
            ...customProperties
        });
    },
    logging: false,
};

export const perfume = new Perfume(perfumeOptions);
```

I'll be following up this issue with a PR soon 😄

**Comment by Zizzamia** [comment]
Amazing, thank you for taking the time to write the PR. Before moving forward I need to change the way `logging` and `sendTiming` is done to support more flexible behavior.

https://github.com/Zizzamia/perfume.js/pull/69#issuecomment-495478385

**Comment by Zizzamia** [comment]
This feature will be available with [v4.4.0](https://github.com/Zizzamia/perfume.js/releases/tag/v4.4.0).
Really appreciated the feedback 🔥🔥🔥
